packages/flutter/lib/src/material/dropdown_menu.dart

```diff
@@ -1125,30 +1157,24 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
       crossAxisUnconstrained: false,
       builder: (BuildContext context, MenuController controller, Widget? child) {
         assert(_initialMenu != null);
-        final bool isCollapsed = widget.inputDecorationTheme?.isCollapsed ?? false;
-        final Widget trailingButton = widget.showTrailingIcon
-            ? Padding(
-                padding: isCollapsed ? EdgeInsets.zero : const EdgeInsets.all(4.0),
-                child: IconButton(
-                  focusNode: _trailingIconButtonFocusNode,
-                  isSelected: controller.isOpen,
-                  constraints: widget.inputDecorationTheme?.suffixIconConstraints,
-                  padding: isCollapsed ? EdgeInsets.zero : null,
-                  icon: widget.trailingIcon ?? const Icon(Icons.arrow_drop_down),
-                  selectedIcon: widget.selectedTrailingIcon ?? const Icon(Icons.arrow_drop_up),
-                  onPressed: !widget.enabled
-                      ? null
-                      : () {
-                          handlePressed(controller);
-                        },
-                ),
-              )
-            : const SizedBox.shrink();
-
-        final Widget leadingButton = Padding(
-          padding: const EdgeInsets.all(8.0),
-          child: widget.leadingIcon ?? const SizedBox.shrink(),
+        final DropdownMenuDecorationBuilder decorationBuilder =
+            widget.decorationBuilder ?? _buildLegacyDecoration;
```

Comment:
nit: do you think if _buildDefaultDecoration will be better? And other places.
```suggestion
            widget.decorationBuilder ?? _buildDefaultDecoration;
```

Reply:
The reason I used 'Legacy' is because my end goal is to not have this logic to be the default in the future.
The default logic I have in mind would replace the IconButton with a simple icon. Doing so we would have to find a way to keep the current behavior which would become optional and marked as "Legacy".

Maybe I should just let this thinking for a future PR :smile: . Let me apply your suggestion, it just mean that `_buildDefaultDecoration` implementation will change in the future. Seems reasonable.